pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/StarTreeIndexReader.java

```diff
@@ -67,17 +66,11 @@ public class StarTreeIndexReader implements Closeable {
    */
   public StarTreeIndexReader(File segmentDirectory, SegmentMetadataImpl segmentMetadata, ReadMode readMode)
       throws IOException, ConfigurationException {
-    Preconditions.checkNotNull(segmentDirectory);
```

Comment:
what's the reason for removing all these? It's guaranteed elsewhere?

Reply:
Yes. It is internal class, thus we can control all the caller behavior. We usually annotate argument that can be null with `@Nullable`, and skip the null check for performance reason